lib/cloud/clients_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientGetAWSSessionIntegration(t *testing.T) {
+	dummyIntegration := "integration-test"
+	dummyRegion := "test-region-123"
+
+	t.Run("without an integration, must return a missing aws integration session provider error", func(t *testing.T) {
+		ctx := context.Background()
+
+		client, err := NewClients()
+		require.NoError(t, err)
+
```

Comment:
Don't we need to close the `Clients`?
```suggestion
		client, err := NewClients()
		require.NoError(t, err)
		t.Cleanup(func() { require.NoError(t, client.Close()) })
```